pubsub_test.go

```diff
@@ -309,6 +315,69 @@ func TestWatch(t *testing.T) {
 	}
 }
 
+func TestPutMany(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	hosts := newNetHosts(ctx, t, 5)
+	vss := make([]*PubsubValueStore, len(hosts))
+	for i := 0; i < len(vss); i++ {
+		fs, err := pubsub.NewFloodSub(ctx, hosts[i])
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		vss[i], err = NewPubsubValueStore(ctx, hosts[i], fs, testValidator{})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	for i := 1; i < len(hosts); i++ {
+		connect(t, hosts[0], hosts[i])
+	}
+
+	const numRuns = 10
+	const numRoutines = 1000 // Note: if changing the numRoutines also change the number of digits in the fmtString
+	const fmtString = "%s-%04d"
+	const baseKey = "/namespace/key"
+
+	for i := 0; i < numRuns; i++ {
+		key := fmt.Sprintf("%s/%d", baseKey, i)
+		doneCh := make(chan error, numRoutines)
+		for j := 0; j < numRoutines; j++ {
+			rtNum := j
+			go func() {
+				err := vss[0].PutValue(ctx, key, []byte(fmt.Sprintf(fmtString, key, rtNum)))
+				doneCh <- err
```

Comment:
It may be possible to simplify this somewhat using an [errgroup](https://godoc.org/golang.org/x/sync/errgroup)

Reply:
done